specutils/analysis/flux.py

```diff
@@ -193,7 +193,17 @@ def _compute_equivalent_width(spectrum, continuum=1, regions=None,
     dx = np.abs(np.diff(spectrum.spectral_axis.bin_edges))
     ew = np.sum((1 - line_flux / cont_flux) * dx)
 
-    return ew.to(spectrum.spectral_axis.unit)
+    # NOTE: uncertainty gets lost during unit translation, so we have to convert first
+    ew = ew.to(spectrum.spectral_axis.unit)
+    if line_flux.uncertainty is not None:
+        # continuum has no uncertainty, here we assume no uncertainty on the bin edges (dx) as well
+        # we need to add the line_flux uncertainties in quadrature for the sum above (which was
+        # summed over the length of dx)
+        ew.uncertainty = np.sqrt((line_flux.uncertainty.value**2)*len(dx)) * dx.unit
```

Comment:
`uncertainty` is not guaranteed to be "stddev" style uncertainties - there's a whole heap of complexity there but for now I would say the thing to do is something like this (which basically follows a similar implementation in `_compute_line_flux` above):
```suggestion
        if isinstance(line_flux.uncertainty, StdDevUncertainty):
            variance_q = line_flux.uncertainty.quantity ** 2
        elif isinstance(line_flux.uncertainty, VarianceUncertainty):
            variance_q = line_flux.uncertainty.quantity
        elif isinstance(line_flux.uncertainty, InverseVariance):
            variance_q = 1/line_flux.uncertainty.quantity
        else:
            warnings.warn(f"Uncertainty type "
                          f"'{line_flux.uncertainty.uncertainty_type}' "
                          f"was not recognized by equivalent_width. Proceeding "
                          f"without uncertainty in result.",
                          AstropyUserWarning)
            variance_q = None
            
        if variance_q is None:
            ew.uncertainty = None
        else:
            ew.uncertainty = np.sqrt(variance_q.value*len(dx)) * dx.unit
            
```

Reply:
Currently `line_flux` just attaches a quantity to the uncertainty attribute as well, so I don't think we we should include the check here unless/until we change that behavior across the board.  

It does appear to be in standard deviation, though, despite my original comment... it just uses variance internally for the computation. See [lines 156-158](https://github.com/astropy/specutils/blob/d1cc4c45f6369c62d76d680f27bb72a47df19235/specutils/analysis/flux.py#L156-L158).

Reply:
Oh gotcha, probably as part of the follow-on PR that adds uncertainty type it would be good to just check the type is stddev in case that changes in the future or something, but not a high priority now that you've clarified that.